relay-kafka/src/config.rs

```diff
@@ -0,0 +1,390 @@
+use std::collections::BTreeMap;
+
+use failure::Fail;
+use serde::{Deserialize, Serialize};
+
+/// Kafka configuration errors.
+#[derive(Fail, Debug)]
+pub enum ConfigError {
+    /// The user referenced a kafka config name that does not exist.
+    #[fail(display = "unknown kafka config name")]
+    UnknownKafkaConfigName,
+    /// The user did not configure 0 shard
+    #[fail(display = "invalid kafka shard configuration: must have shard with index 0")]
+    InvalidShard,
+}
+
+/// Define the topics over which Relay communicates with Sentry.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub enum KafkaTopic {
+    /// Simple events (without attachments) topic.
+    Events,
+    /// Complex events (with attachments) topic.
+    Attachments,
+    /// Transaction events topic.
+    Transactions,
+    /// Shared outcomes topic for Relay and Sentry.
+    Outcomes,
+    /// Override for billing critical outcomes.
+    OutcomesBilling,
+    /// Session health updates.
+    Sessions,
+    /// Any metric that is extracted from sessions.
+    MetricsSessions,
+    /// Any metric that is extracted from transactions.
+    MetricsTransactions,
+    /// Profiles
+    Profiles,
+    /// ReplayEvents, breadcrumb + session updates for replays
+    ReplayEvents,
+    /// ReplayRecordings, large blobs sent by the replay sdk
+    ReplayRecordings,
+}
+
+impl KafkaTopic {
+    /// Returns iterator over the variants of [`KafkaTopic`].
+    /// It will have to be adjusted if the new variants are added.
+    pub fn iter() -> std::slice::Iter<'static, KafkaTopic> {
+        use KafkaTopic::*;
+        static TOPICS: [KafkaTopic; 11] = [
+            Events,
+            Attachments,
+            Transactions,
+            Outcomes,
+            OutcomesBilling,
+            Sessions,
+            MetricsSessions,
+            MetricsTransactions,
+            Profiles,
+            ReplayEvents,
+            ReplayRecordings,
+        ];
+        TOPICS.iter()
+    }
+}
+
+/// Configuration for topics.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(default)]
+pub struct TopicAssignments {
```

Comment:
To separate concerns, it would be beneficial if the `relay-kafka` crate doesn't know about the topic assignments. For instance, we could leave the `TopicAssignment` type here, but move `TopicAssignments` and `KafkaTopic` back into `relay-config` or wherever it's best defined. That way, we don't have to rebuild it every time we add a new topic. 

I'm happy to merge the PR as-is though, and follow up with this later.

Reply:
Agreed! I will followup on this